src/db/error.rs

```diff
@@ -85,13 +86,35 @@ impl From<Context<DbErrorKind>> for DbError {
     }
 }
 
+// XXX: maybe not worth the effort vs
+// DbErrorKind::{DieselConnectionError, SpannerConnectionError}
+#[derive(Debug)]
+pub enum ConnectionError {
+    Diesel(diesel::result::ConnectionError),
+    Spanner(String),
+}
+
+impl fmt::Display for ConnectionError {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        match self {
```

Comment:
Aren't these basically the same thing? Do you need the match?

Reply:
it's still needed to break down the field of the enum, but that could be clearer